src/pages/t/[slug]/edit.tsx

```diff
@@ -375,7 +375,7 @@ export default function EditProfilePage({ slug }: { slug: string }) {
   };
 
   useEffect(() => {
-    if (userInfo && slug !== userInfo?.username) {
+    if (!userInfo || (userInfo && slug !== userInfo?.username)) {
```

Comment:
this does not solve the issue; better to reroute to `/` for non logged out user; plus would suggest you to check that through `useSession`

Reply:
ok will do. Also fixed the pnpm lock issue from my side. misunderstood the contributing guidelines.